tests/test_repro.py

```diff
@@ -1404,3 +1393,58 @@ def test(self):
 
         expected_metrics_display = "{}: {}".format(metrics_file, metrics_value)
         self.assertIn(expected_metrics_display, self._caplog.text)
+
+
+# Pytest style tests
+
+
+@pytest.fixture
+def foo_copy(repo_dir, dvc):
+    stages = dvc.add(repo_dir.FOO)
+    assert len(stages) == 1
+    foo_stage = stages[0]
+    assert foo_stage is not None
+
+    fname = "foo_copy"
+    stage_fname = fname + ".dvc"
+    dvc.run(
+        fname=stage_fname,
+        outs=[fname],
+        deps=[repo_dir.FOO, repo_dir.CODE],
+        cmd="python {} {} {}".format(repo_dir.CODE, repo_dir.FOO, fname),
+    )
+    return {"fname": fname, "stage_fname": stage_fname}
+
+
+def test_dvc_formatting_retained(dvc, foo_copy):
+    root = Path(dvc.root_dir)
+    stage_file = root / foo_copy["stage_fname"]
+
+    # Add comments and custom formatting to stage file
+    lines = list(map(_format_dvc_line, stage_file.read_text().splitlines()))
+    lines.insert(0, "# Starting comment")
+    stage_text = "".join(l + "\n" for l in lines)
```

Comment:
just to double check. is it possible that on some platforms yaml is dumped with `\r` or something else? In this case md5 below won't match. I'm not sure how big is the problem.

Reply:
Not sure, but we use literal comparisons in other tests too.